### PR TITLE
feat: Connect documents page to cms

### PR DESCRIPTION
**Pull request by abbyoung**
# SC-1439

* Queries CMS database for documents page, sections, and documents.
* Updates tests

## Reviewer notes

* This PR relies on ussf-portal-cms branch of the same name, `sc-1435-documents-page` in order to work. 
* **This PR must be merged after** https://github.com/USSF-ORBIT/ussf-portal-cms/pull/253
* e2e tests can be found at https://github.com/USSF-ORBIT/ussf-portal/pull/191

* There is currently no hard limit on the db that prevents us from creating more than one documents page. The portal will display the *first* documents page created. 

## Setup

* Check out cms branch `sc-1435-documents-page` and this branch
* Run both the cms and the client locally
* In the CMS, create a document, a document section, and a documents page. 
* In the portal, visit `/ussf-documentation` and confirm you see the cms content displayed

### As the original developer, I have

- [ ] Met the acceptance criteria, or will meet them in subsequent PRs or stories
  - ... 
- [ ] Created new stories in Storybook if applicable
  - [ ] Checked that all Storybook accessibility checks are passing
- [ ] Created/modified automated unit tests in Jest
  - [ ] Including jest-axe checks when UI changes
- [ ] Created/modified automated E2E tests in Cypress
  - [ ] Including cypress-axe checks when UI changes
  - [ ] Checked that the E2E test build is not failing
- Performed [a11y testing](https://github.com/trussworks/accessibility/blob/master/sample_a11y_testing_process.md):
  - [ ] Checked responsiveness in mobile, tablet, and desktop
  - [ ] Checked keyboard navigability
  - [ ] Tested with [VoiceOver](https://dequeuniversity.com/screenreaders/voiceover-keyboard-shortcuts) in Safari
  - [ ] Checked VO's [rotor menu](https://github.com/trussworks/accessibility/blob/master/README.md#how-to-use-the-rotor-menu) for landmarks, page heading structure and links
  - [ ] Used a browser a11y tool to check for issues (WAVE, axe, ANDI or Accessibility addon tab for Storybook)
- [ ] Requested a design review for user-facing changes
- For any new migrations/schema changes:
  - [ ] Followed guidelines for zero-downtime deploys

### As code reviewer(s), I have

- [ ] Pulled this branch locally and tested it
- [ ] Reviewed this code and left comments
- [ ] Checked that all code is adequately covered by tests
  - [ ] Checked that the E2E test build is not failing
- [ ] Made it clear which comments need to be addressed before this work is merged
- [ ] Considered marking this as accepted even if there are small changes needed
- Performed [a11y testing](https://github.com/trussworks/accessibility/blob/master/sample_a11y_testing_process.md):
  - [ ] Checked responsiveness in mobile, tablet, and desktop
  - [ ] Checked keyboard navigability
  - [ ] Tested with [VoiceOver](https://dequeuniversity.com/screenreaders/voiceover-keyboard-shortcuts) in Safari
  - [ ] Checked VO's [rotor menu](https://github.com/trussworks/accessibility/blob/master/README.md#how-to-use-the-rotor-menu) for landmarks, page heading structure and links
  - [ ] Used a browser a11y tool to check for issues (WAVE, axe, ANDI or Accessibility addon tab for Storybook)

### As a designer reviewer, I have

- [ ] Checked in the design translated visually
- [ ] Checked behavior
- [ ] Checked different states (empty, one, some, error)
- [ ] Checked for landmarks, page heading structure, and links
- [ ] Tried to break the intended flow
- Performed [a11y testing](https://github.com/trussworks/accessibility/blob/master/sample_a11y_testing_process.md):
  - [ ] Checked responsiveness in mobile, tablet, and desktop
  - [ ] Checked keyboard navigability
  - [ ] Tested with [VoiceOver](https://dequeuniversity.com/screenreaders/voiceover-keyboard-shortcuts) in Safari
  - [ ] Checked VO's [rotor menu](https://github.com/trussworks/accessibility/blob/master/README.md#how-to-use-the-rotor-menu) for landmarks, page heading structure and links
  - [ ] Used a browser a11y tool to check for issues (WAVE, axe, ANDI or Accessibility addon tab for Storybook)

### As a test user, I have

- Run through the [Test Script](hhttps://docs.google.com/spreadsheets/d/1eV8UK0aJZ0qrzjnXf5SJ_uRiV7bpsj3yrhRFEkjOvV4/edit?usp=sharing):
  - [ ] On commercial internet in IE11
  - [ ] On commercial internet in Firefox
  - [ ] On commercial internet in Chrome
  - [ ] On commercial internet in Safari
  - [ ] On NIPR in IE11
  - [ ] On NIPR in Firefox
  - [ ] On NIPR in Chrome
  - [ ] On NIPR in Safari
  - [ ] On a mobile device in Firefox
  - [ ] On a mobile device in Chrome
  - [ ] On a mobile device in Safari
- [ ] Added any anomalous behavior to this PR

---

### CODE REVIEW

File: src/pages/ussf-documentation.tsx [hunk 1]
```diff
@@ -1,83 +1,59 @@
...
-          />
-        </Grid>
+        {sections &&
+          sections.map((s: DocumentSectionType) => (
+            <Grid col={8} className={styles.accordion} key={s.id}>
+              <Accordion
+                bordered
+                items={[
+                  {
+                    title: s.title,
```
Comment (jcbcapps) [inline src/pages/ussf-documentation.tsx:33]: Dark mode styles weren't showing properly here. I moved `className={styles.accordion}` to the `Accordion` and that fixed it!
```suggestion
            <Grid col={8} key={s.id}>
              <Accordion
                className={styles.accordion}
                bordered
                items={[
                  {
                    title: s.title,
```

File: src/pages/ussf-documentation.tsx [hunk 2]
```diff
@@ -1,83 +1,59 @@
...
+                    expanded: true,
```
Comment (jcbcapps) [inline src/pages/ussf-documentation.tsx:34]: I'm not 100% sure of this, but I think only the first section is supposed to be expanded by default? If so, you could probably get the index from the `.map()` and use that as a check for `true` or `false` here.

**Review by jcbcapps — APPROVED** [review]
Looks great! I left a couple of comments, but I don't think they are problematic.

File: src/pages/ussf-documentation.tsx (hunk 2 above)
Comment (abbyoung) [reply src/pages/ussf-documentation.tsx:34]: great catch! thanks!

File: src/pages/ussf-documentation.tsx (hunk 1 above)
Comment (abbyoung) [reply src/pages/ussf-documentation.tsx:33]: thank you!

File: src/styles/pages/ussf-documentation.module.scss [hunk 3]
```diff
@@ -1,40 +1,50 @@
 @import 'styles/uswdsDependencies';
 @import 'styles/sfds/sfdsDependencies';
 
-.accordion {
-  > h3 > button {
-    background-color: var(--accordion-btn-bg);
-    color: var(--accordion-title);
+.accordionGrid {
+  :global(.usa-accordion--bordered .usa-accordion__heading) {
+    margin-top: 0.5rem;
```
Comment (abbyoung) [inline src/styles/pages/ussf-documentation.module.scss:6]: This looks like a lot of changes, but I just needed to wrap everything in a new parent class, `accordionGrid`

**Review by jcbcapps — CHANGES_REQUESTED** [review]
If there is not a Documents Page in the CMS and I try to navigate to `/ussf-documentation` the app crashes. Just a thought, but maybe we shouldn't display the `USSF documentation` option in the side nav if there is no Document Page in the CMS? Or maybe we just leave the page blank? No idea, but thought it worth mentioning!

**Review by gidjin — APPROVED** [review]
Left a couple suggestions but they are in the tests files so I'm good without a re-review.

File: src/pages/ussf-documentation.tsx [hunk 4]
```diff
@@ -1,92 +1,154 @@
 import React from 'react'
 import { Accordion, Grid } from '@trussworks/react-uswds'
-import { AccordionItemProps } from '@trussworks/react-uswds/lib/components/Accordion/Accordion'
+import { InferGetServerSidePropsType } from 'next'
+import { useFlags } from 'launchdarkly-react-client-sdk'
```
Comment (abbyoung) [inline src/pages/ussf-documentation.tsx:4]: Using `withLDConsumer` breaks the `getLayout` function, so we're using hooks

**Review by jcbcapps — APPROVED** [review]
LGTM!

**Review by gidjin — APPROVED** [review]
LGTM!

**Pull request merged**